easybuild/easyblocks/p/python.py

```diff
@@ -192,6 +192,29 @@ def patch_step(self, *args, **kwargs):
             # Ignore user site dir. -E ignores PYTHONNOUSERSITE, so we have to add -s
             apply_regex_substitutions('configure', [(r"(PYTHON_FOR_BUILD=.*-E)'", r"\1 -s'")])
 
+        # If we filter out LD_LIBRARY_PATH (not unusual when using rpath), ctypes is not able to dynamically load
+        # libraries installed with EasyBuild (see https://github.com/EESSI/software-layer/issues/192).
+        # ctypes is using GCC (and therefore LIBRARY_PATH) to figure out the full location but then only returns the
+        # soname, instead let's return the full path in this particular scenario
+        filtered_env_vars = build_option('filter_env_vars') or []
+        if 'LD_LIBRARY_PATH' in filtered_env_vars and 'LIBRARY_PATH' not in filtered_env_vars:
+            ctypes_util_py = os.path.join("Lib", "ctypes", "util.py")
+            orig_gcc_so_name = None
+            # Let's do this incrementally since we are going back in time
+            if LooseVersion(self.version) >= "3.9.1":
+                # From 3.9.1 to at least v3.12.4 there is only one match for this line
+                orig_gcc_so_name = "_get_soname(_findLib_gcc(name)) or _get_soname(_findLib_ld(name))"
+            if orig_gcc_so_name:
+                orig_gcc_so_name_regex = r'(\s*)' + re.escape(orig_gcc_so_name) + r'(\s*)'
+                updated_gcc_so_name = (
+                    "_findLib_gcc(name) or _findLib_ld(name)"
```

Comment:
Given that we are effectively dropping the use of `_get_soname()` we should probably add an explanation of why. You already have a good explanation in our Slack chat about this
```
_get_soname() uses objdump to get the SONAME field from the shared object file. The presence or absence of the SONAME field in the ELF header of a shared library is influenced by how the library is compiled and linked. For manually built libraries we may be lacking this field.
```

Reply:
I forgot this was my PR, done in https://github.com/easybuilders/easybuild-easyblocks/pull/3352/commits/1ee17c0f7726c69e97442f53c65c5f041d65c94f